### ISSUE TITLE
Batching data? 

### ISSUE THREAD

**Issue by Dadatata-JZ** [takeuchi-lab/MS-DA-MIL-CNN#1]
Hi there, 

Thank you for publishing this inspiring research outcome and the associated implementation. I went through your implementation and realized that it seemed you loaded all training images to a tensor and trained on it. I am curious how large was your dataset. Given a 16gb GPU, I am impressed by that fact that there was no batching involved unless the dataset is not large. 

If there was a batching or similar approach, please redirect me to the code part. 
Thank you in advance.

**Comment by Dadatata-JZ** [comment]
@takeuchi-write

**Comment by takeuchi-write** [comment]
Thank you for a comment.
In our source code, a single bag of images corresponds to a batch. 

In the following code, 

```
def train(model, device, loss_fn, optimizer, train_data, DArate):
    model.train() # train mode
    train_class_loss = 0.0
    train_domain_loss = 0.0
    correct_num = 0
    # shuffle bag
    random.shuffle(train_data)
    for data in train_data:
        # loading data
        input_tensor, class_label, domain_label = utils.data_load(data)
```

'data' includes only file paths and label data, and it does not contain image tensors.
Then, image tensors in a bag (for a batch) are returned by utils.data_load(data) and input into the model.
